components/gitpod-protocol/src/workspace-instance.ts

```diff
@@ -97,8 +97,12 @@ export type WorkspaceInstancePhase =
     // a workspace. This phase is usually accompanied by an error.
     | "unknown"
 
+    // Building means that we are building the Docker image for the workspace. A workspace will enter this phase only
+    // if an image build is required for that workspace.
+    | "building"
+
     // Preparing means that we haven't actually started the workspace instance just yet, but rather
-    // are still preparing for launch. This means we're building the Docker image for the workspace.
+    // are still preparing for launch.
```

Comment:
I'd switch order here, to denote the ordering:
 1. preparing: Preparing means that we haven't sent any requests to ws-manager, yet, but are still processing the request to start a workspace, e.g. context parsing.
 2. imagebuilding: Imagebuilding means...

Reply:
The reason to have it this was is that we're looking to move `imagebuilding` into `ws-manager`, so that `preparing` is the only phase left on WebApp's side, and will cover everything from "context parsing -> ws-manager req sent".